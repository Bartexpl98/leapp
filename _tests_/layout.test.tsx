import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';
import React from "react";

// Mock Next components
jest.mock("next/link", () => ({ children }: { children: React.ReactNode }) => <div>{children}</div>);
jest.mock("next/image", () => ({ src, alt }: { src: string; alt: string }) => <img src={src} alt={alt} />);

// Import Page component
import Page from "@/app/page";

describe("Page Component", () => {
  it("renders welcome text", () => {
    render(<Page />);

    // Use regex to match partial text if strong breaks it
    expect(
      screen.getByText((content, element) =>
        content.includes("Welcome to Acme")
      )
    ).toBeInTheDocument();
  });

  it("renders login link", () => {
    render(<Page />);
    expect(screen.getByText("Log in")).toBeInTheDocument();
  });

  it("renders image", () => {
    render(<Page />);
    expect(screen.getByAltText(/Screenshots of the dashboard project/i)).toBeInTheDocument();
  });
});
