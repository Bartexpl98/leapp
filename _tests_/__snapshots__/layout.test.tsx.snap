// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`RootLayout matches snapshot 1`] = `
<div>
  <div>
    Mock Providers 
    <div>
      Mock Header
    </div>
    <div
      class="flex min-h-screen flex-col md:flex-row"
    >
      <aside
        class="hidden md:block md:w-64 bg-gray-100 p-4"
      >
        <div>
          Mock SideNav
        </div>
      </aside>
      <main
        class="flex-1 p-4 sm:p-6"
      >
        <div>
          Snapshot Child
        </div>
      </main>
    </div>
  </div>
</div>
`;
